tests/integration/test_security_report.py

```diff
@@ -157,12 +157,86 @@ def test_csp_violation_reports_with_processing(
     events.append(event)
-    events_consumer.assert_empty()
 
+    events_consumer.assert_empty()
+    assert len(events), 2
     assert any(d for d in events if d["csp"]["disposition"] == "enforce")
     assert any(d for d in events if d["csp"]["disposition"] == "report")
 
 
+def test_deprication_reports_with_processing(
+    mini_sentry,
+    relay_with_processing,
+    events_consumer,
+):
+    # UA parsing introduces higher latency in debug mode
+    events_consumer = events_consumer(timeout=10)
+
+    proj_id = 42
+    relay = relay_with_processing()
+    mini_sentry.add_full_project_config(proj_id)
+
+    reports = [
+        {
+            "age": 0,
+            "body": {
+                "blockedURL": "https://example.com/tst/media/7_del.png",
+                "disposition": "report",
+                "documentURL": "https://example.com/tst/test_frame.php?ID=229&hash=da964209653e467d337313e51876e27d",
+                "effectiveDirective": "img-src",
+                "lineNumber": 9,
+                "originalPolicy": "default-src 'none'; report-to endpoint-csp;",
+                "referrer": "https://example.com/test229/",
+                "sourceFile": "https://example.com/tst/test_frame.php?ID=229&hash=da964209653e467d337313e51876e27d",
+                "statusCode": 0,
+            },
+            "type": "csp-violation",
+            "url": "https://example.com/tst/test_frame.php?ID=229&hash=da964209653e467d337313e51876e27d",
+            "user_agent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36",
+        },
+        {
+            "type": "deprecation",
+            "age": 10,
+            "url": "https://example.com/",
+            "user_agent": "BarBrowser/98.0 (Mozilla/5.0 compatiblish)",
+            "body": {
+                "id": "websql",
+                "anticipatedRemoval": "1/1/2020",
+                "message": "WebSQL is deprecated and will be removed in Chrome 97 around January 2020",
+                "sourceFile": "https://example.com/index.js",
+                "lineNumber": 1234,
+                "columnNumber": 42,
+            },
+        },
+    ]
+
+    try:
+        relay.send_security_report(
+            project_id=proj_id,
+            content_type="application/reports+json; charset=utf-8",
+            payload=reports,
+            release="01d5c3165d9fbc5c8bdcf9550a1d6793a80fc02b",
+            environment="production",
+        )
+
+        events = []
+        event, _ = events_consumer.get_event()
+        events.append(event)
+
+        # We will discard the unsupported "deprecation" report type.
+        assert len(mini_sentry.test_failures) > 0
+        assert {str(e) for _, e in mini_sentry.test_failures} == {
+            "Relay sent us event: failed to extract security report: event filtered with reason: InvalidCsp",
+            "Relay sent us event: dropped envelope: internal error",
```

Comment:
Why is this error okay, there shouldn't be internal errors because of correctly handled CSP reports?

Reply:
From what I see in the code, we let this particular error fall through and at the end of the processing we just drop the managed envelope, which produces this error.  This seems like expected  behaviour.

Reply:
But that means we will get Sentry errors with `dropped envelope: internal error` *and* outcomes with outcome `Invalid(Internal)` don't think this is good.

Reply:
ok, revisit this part in [137abbd](https://github.com/getsentry/relay/pull/3295/commits/137abbd96a48788e620fd3eebb70e5d7c46a70ca)